src/config.js

```diff
@@ -521,6 +524,75 @@ export default class Config {
     }
   }
 
+  async findManifest(dir: string, isRoot: boolean): Promise<?Manifest> {
+    for (const registry of registryNames) {
+      const manifest = await this.tryManifest(dir, registry, isRoot);
+
+      if (manifest) {
+        return manifest;
+      }
+    }
+
+    return null;
+  }
+
+  async findWorktree(initial: string): Promise<?string> {
+    let previous = null;
```

Comment:
Hm I was thinking of implementing this check inside `Config#init`, would that be fine with you? The problem with `resolveWorkspaces` is that it doesn't have to be executed in every case (for example, when adding a dependency we don't use it, since we only care about the current workspace and the base worktree).